### PR TITLE
Fix for coverage collection.

### PR DESCRIPTION
**Pull request by yifanyang**
Fix failing task `:tools:measurement:apksize:preFirestoreAggressiveBuild`.
Make coverage task depend on only firebase product check tasks.

The tasks dependency graph previously looks like:
```
:tools:measurement:coverage:uploadCoverageMeasurements
\--- :tools:measurement:coverage:generateCoverageMeasurements
     +--- :fiamui-app:checkCoverage
     |    \--- :fiamui-app:check ..>
     +--- :firebase-common:checkCoverage
     |    \--- :firebase-common:check ..>
     +--- :firebase-database:checkCoverage
     |    \--- :firebase-database:check ..>
     +--- :firebase-database-collection:checkCoverage
     |    \--- :firebase-database-collection:check ..>
     +--- :firebase-datatransport:checkCoverage
     |    \--- :firebase-datatransport:check ..>
     +--- :firebase-firestore:checkCoverage
     |    \--- :firebase-firestore:check ..>
     +--- :firebase-functions:checkCoverage
     |    \--- :firebase-functions:check ..>
     +--- :firebase-inappmessaging-display:checkCoverage
     |    \--- :firebase-inappmessaging-display:check ..>
     +--- :firebase-storage:checkCoverage
     |    \--- :firebase-storage:check ..>
     +--- :protolite-well-known-types:checkCoverage
     |    \--- :protolite-well-known-types:check ..>
     +--- :tools:checkCoverage
     |    \--- :tools:check ..>
     +--- :transport:checkCoverage
     |    \--- :transport:check ..>
     +--- :firebase-common:ktx:checkCoverage
     |    \--- :firebase-common:ktx:check ..>
     +--- :firebase-firestore:ktx:checkCoverage
     |    \--- :firebase-firestore:ktx:check ..>
     +--- :firebase-functions:ktx:checkCoverage
     |    \--- :firebase-functions:ktx:check ..>
     +--- :firebase-storage:test-app:checkCoverage
     |    \--- :firebase-storage:test-app:check ..>
     +--- :tools:errorprone:checkCoverage
     |    \--- :tools:errorprone:check ..>
     +--- :tools:errorprone:jacocoTestReport
     |    \--- :tools:errorprone:classes ..>
     +--- :tools:lint:checkCoverage
     |    \--- :tools:lint:check ..>
     +--- :tools:lint:jacocoTestReport
     |    +--- :tools:lint:classes ..>
     |    \--- :tools:lint:compileKotlin
     +--- :tools:measurement:checkCoverage
     |    \--- :tools:measurement:check ..>
     +--- :transport:transport-api:checkCoverage
     |    \--- :transport:transport-api:check ..>
     +--- :transport:transport-backend-cct:checkCoverage
     |    \--- :transport:transport-backend-cct:check ..>
     +--- :transport:transport-runtime:checkCoverage
     |    \--- :transport:transport-runtime:check ..>
     +--- :tools:measurement:apksize:checkCoverage
     |    \--- :tools:measurement:apksize:check ..>
     \--- :tools:measurement:coverage:checkCoverage
          \--- :tools:measurement:coverage:check ..>
```

Now it looks like:
```
:tools:measurement:coverage:uploadCoverageMeasurements
\--- :tools:measurement:coverage:generateCoverageMeasurements
     +--- :firebase-common:checkCoverage
     |    \--- :firebase-common:check ..>
     +--- :firebase-database:checkCoverage
     |    \--- :firebase-database:check ..>
     +--- :firebase-database-collection:checkCoverage
     |    \--- :firebase-database-collection:check ..>
     +--- :firebase-datatransport:checkCoverage
     |    \--- :firebase-datatransport:check ..>
     +--- :firebase-firestore:checkCoverage
     |    \--- :firebase-firestore:check ..>
     +--- :firebase-functions:checkCoverage
     |    \--- :firebase-functions:check ..>
     +--- :firebase-inappmessaging-display:checkCoverage
     |    \--- :firebase-inappmessaging-display:check ..>
     +--- :firebase-storage:checkCoverage
     |    \--- :firebase-storage:check ..>
     +--- :protolite-well-known-types:checkCoverage
     |    \--- :protolite-well-known-types:check ..>
     +--- :firebase-common:ktx:checkCoverage
     |    \--- :firebase-common:ktx:check ..>
     +--- :firebase-firestore:ktx:checkCoverage
     |    \--- :firebase-firestore:ktx:check ..>
     +--- :firebase-functions:ktx:checkCoverage
     |    \--- :firebase-functions:ktx:check ..>
     +--- :transport:transport-api:checkCoverage
     |    \--- :transport:transport-api:check ..>
     +--- :transport:transport-backend-cct:checkCoverage
     |    \--- :transport:transport-backend-cct:check ..>
     \--- :transport:transport-runtime:checkCoverage
          \--- :transport:transport-runtime:check ..>
```

### CODE REVIEW

File: tools/measurement/coverage/coverage.gradle [hunk 1]
```diff
@@ -21,7 +21,11 @@ task generateCoverageMeasurements(type: GenerateMeasurementsTask) {
     description 'Runs checkCoverage task in all projects and calculates coverage percents.'
     group 'Measurements'
 
-    dependsOn rootProject.allprojects.collect { it.tasks.withType(JacocoReport) }.flatten()
+    dependsOn rootProject.allprojects.findAll {
+        it.path.startsWith(':firebase-')
```
Comment (yifanyang) [inline tools/measurement/coverage/coverage.gradle]: The intention here is to only depend on check tasks of firebase products instead of all. Is there a more reliant way to distinguish firebase products from other utility subprojects?

Comment (allisonbm92) [reply tools/measurement/coverage/coverage.gradle]: You can check if the projects are using the Firebase library plugin. This plugin encapsulates some of the release and testing logic, and we only use it on the actual Firebase projects.

`project.extensions.findByType(FirebaseLibraryExtension.class) != null` should do the trick.

Comment (yifanyang) [reply tools/measurement/coverage/coverage.gradle]: Looks like all below projects have this plugin:
```
project ':firebase-common'
project ':firebase-database'
project ':firebase-database-collection'
project ':firebase-datatransport'
project ':firebase-firestore'
project ':firebase-functions'
project ':firebase-inappmessaging-display'
project ':firebase-storage'
project ':protolite-well-known-types'
project ':firebase-common:ktx'
project ':firebase-firestore:ktx'
project ':firebase-functions:ktx'
project ':transport:transport-api'
project ':transport:transport-backend-cct'
project ':transport:transport-runtime'
```
by running
```
rootProject.allprojects.findAll { it.extensions.findByType(FirebaseLibraryExtension) != null }.forEach { println it }
```

Anyway to exclude `:protolite-well-known-types`, `:transport:transport-api`, `:transport:transport-backend-cct`, and `:transport:transport-runtime`?

Comment (allisonbm92) [reply tools/measurement/coverage/coverage.gradle]: I think we should include those. The transport ones are part of Firelog. Any reason for excluding?

Comment (yifanyang) [reply tools/measurement/coverage/coverage.gradle]: Done. Discussed offline.

File: tools/measurement/apksize/apksize.gradle [hunk 2]
```diff
@@ -21,6 +21,10 @@ android {
     flavorDimensions "apkSize"
 }
 
+dependencies {
+    implementation("com.google.android.gms:play-services-basement:16.1.0")
```
Comment (allisonbm92) [inline tools/measurement/apksize/apksize.gradle]: Why is this needed?

Comment (yifanyang) [reply tools/measurement/apksize/apksize.gradle]: Removed.

We used to see below failure from coverage tasks log (https://pantheon.corp.google.com/storage/browser/fireescape/pr-logs/pull/firebase_firebase-android-sdk/489/coverage-metrics-upload/1135971121242312705/?project=fireescape-c4819&folder=true&organizationId=true).
```
Task :tools:measurement:apksize:preFirestoreAggressiveBuild FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tools:measurement:apksize:preFirestoreAggressiveBuild'.
> Android dependency 'com.google.android.gms:play-services-basement' has different version for the compile (16.0.1) and runtime (16.1.0) classpath. You should manually set the same version via DependencyResolution
```

But looks like #489 fixed it, somehow.
